### PR TITLE
785 colsubsetexpr refactor

### PR DESCRIPTION
**Pull request by gmbecker**
Not sure what happened to the original PR. I guess it got destroyed when I rebased on main after doing the inline commit thing (which sadly messed everything up by making things out of sink)

Sorry about but I think the (now lost) comments are addressed here...

### CODE REVIEW

**Comment by gmbecker** [comment]
@jcheng5

File: tests/testthat/test_utils.R [hunk 1]
```diff
@@ -1,6 +1,7 @@
 context("Checking utility functions")
 
 test_that("func_takes works with different inputs", {
+    func_takes <- rtables:::func_takes
```
Comment (Melkiades) [inline tests/testthat/test_utils.R:4]: remember to take this out!

**Review by Melkiades — COMMENTED** [review]
Lgtm! Thank you, Gabe, for these additions. Only remember to take out the `:::` and add NEWS entry ;)

**Review by Melkiades — COMMENTED** [review]
Still to do:

- [ ] NEWS entry
- [ ] remove `:::`
- [ ] make it pass the PRs tests and keep the coverage up with proper testing
- [ ] test it downstream with tlg catalog

**Comment by gmbecker** [comment]
This is subsumed into #865

**Pull request closed without merging**